Login Screen

vanyalaptii opened the pull request:
Was created SignInView and Firebase database connected

vanyalaptii committed:
Login Screen added

Was created SignInView and Firebase database connected

OksanaFedorchuk commented on ReFix/Views/SignInView.swift:
```diff
@@ -0,0 +1,79 @@
+                .frame(height: 150)
+                
+                //TODO: navigation to forgot password
+                Button(action: /*@START_MENU_TOKEN@*/{}/*@END_MENU_TOKEN@*/, label: {
```
а що це таке? 
/*@START_MENU_TOKEN@*/{}/*@END_MENU_TOKEN@*/

vanyalaptii replied:
У мене там нічого нема просто пусті дужки {}

OksanaFedorchuk commented on ReFix/Views/SignInView.swift:
```diff
@@ -0,0 +1,79 @@
+    var body: some View {
+        ZStack {
+            
+            Color(.secondarySystemBackground)
```
створюй бекграунд через .backgound modifier

vanyalaptii replied:
Зробив

OksanaFedorchuk commented:
а де Гугл пліст? в мене червоний і помилка
Build input file cannot be found: '/Users/Downloads/GoogleService-Info.plist'. Did you forget to declare this file as an output of a script phase or custom build rule which produces it?

OksanaFedorchuk commented on ReFix/Views/SignInView.swift:
```diff
@@ -0,0 +1,79 @@
+            
+            VStack {
+                
+                Image("ReFixLogo")
```
потрібно декомпозувати в"ю, щоб було з першого погляду зрозуміло і видно що там

vanyalaptii replied:
Зробив

OksanaFedorchuk commented on ReFix/Views/SignInView.swift:
```diff
@@ -0,0 +1,79 @@
+                                .offset(x: 7)
+                        }
+                }
+                .frame(height: 150)
```
фрейм краще виставляти не хардкодом, а задавати мін макс тощо

vanyalaptii replied:
Маю питання з цього приводу

OksanaFedorchuk commented on ReFix/Views/SignInView.swift:
```diff
@@ -0,0 +1,79 @@
+                Button {} label: {
+                    ZStack{
+                        RoundedRectangle(cornerRadius: 10)
+                        Text("Увійти")
```
якщо це бекграунд для тексту, то зроби як бекграунд. Але взагалі кнопки краще налаштовувати через .buttonStyle Спробуй заюзати якісь стандартні стилі
https://www.rootstrap.com/blog/swiftui-button-a-complete-tutorial#:~:text=SwiftUI%20Button%20Style&text=SwiftUI%20provides%20several%20built%2Din,use%20the%20buttonStyle()%20modifier.

vanyalaptii replied:
Поки виправив через бекграунд, про стилі почитаю

OksanaFedorchuk commented on ReFix/Views/SignInView.swift:
```diff
@@ -0,0 +1,79 @@
+                    }
+                    .frame(width: 90, height: 40)
+                }
+                .offset(y: 150)
```
не юзай офсети для відступів. Щоб регулювати положення елементів, юзай Spacer(), (spacing: ) - on stacks, or .padding

vanyalaptii replied:
Виправив

vanyalaptii committed:
Made corrections after the review

vanyalaptii committed:
Google info added

vanyalaptii committed:
Autentification view updates

1. Decomposed views
2. The row text field was renamed  to customTextField and customTextFieldWithIcon.
3. .frame modifier changed with parameters maxHeight and minHeight.
4. Dismissing the keyboard is implemented using a @FocusState parameter.
5.Added ClearButtonViewModifier for a TextField.
6.Using buttonStyle() modifier for buttons.
7. SignInView and SighUpView are combined into one AuthView that uses  authState parameter to switch them.

vanyalaptii committed:
Autentification view updates

1. Decomposed views
2. The row text field was renamed  to customTextField and customTextFieldWithIcon.
3. .frame modifier changed with parameters maxHeight and minHeight.
4. Dismissing the keyboard is implemented using a @FocusState parameter.
5.Added ClearButtonViewModifier for a TextField.
6.Using buttonStyle() modifier for buttons.
7. SignInView and SighUpView are combined into one AuthView that uses  authState parameter to switch them.

vanyalaptii committed:
Merge branch 'feature/LoginScreenView' of https://github.com/vanyalaptii/ReFix into feature/LoginScreenView

vanyalaptii committed:
Autentification view separated

1. Autentification view separated to a different views(SignInView and SignUpView) and managed in AuthView.
2. Added AuthViewModel for AuthView.

vanyalaptii committed:
Project folders reorganized

1. Project folders reorganized to modules that contains views, subviews and viewmodels.
2. Background removed from AuthView and added to SignInView and SignUpView
